src/pages/workspace/members/WorkspaceOwnerChangeCheckPage.tsx

```diff
@@ -0,0 +1,110 @@
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as WorkspaceSettingsUtils from '@libs/WorkspacesSettingsUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import type {WithPolicyOnyxProps} from '@pages/workspace/withPolicy';
+import withPolicy from '@pages/workspace/withPolicy';
+import * as PolicyActions from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {PersonalDetailsList} from '@src/types/onyx';
+
+type WorkspaceOwnershipChangeChecksOnyxProps = {
+    /** Personal details of all users */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
+};
+
+type WorkspaceMemberDetailsPageProps = WithPolicyOnyxProps &
+    WorkspaceOwnershipChangeChecksOnyxProps &
+    StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.OWNER_CHANGE_CHECK>;
+
+function WorkspaceOwnerChangeCheckPage({route, personalDetails, policy}: WorkspaceMemberDetailsPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const policyID = route.params.policyID;
+    const accountID = route.params.accountID;
+    const error = route.params.error;
+
+    useEffect(() => {
+        if (!policy) {
+            return;
+        }
+
+        if (!policy?.errorFields?.changeOwner) {
+            Navigation.navigate(ROUTES.WORKSPACE_MEMBER_DETAILS.getRoute(policyID, accountID));
+            return;
+        }
+
+        const keys = Object.keys(policy.errorFields.changeOwner);
+
+        if (keys && keys.length > 0 && keys[0] !== CONST.POLICY.OWNERSHIP_ERRORS.NO_BILLING_CARD) {
+            Navigation.navigate(ROUTES.WORKSPACE_OWNER_CHANGE_CHECK.getRoute(policyID, accountID, keys[0] as ValueOf<typeof CONST.POLICY.OWNERSHIP_ERRORS>));
+        }
+    }, [accountID, policy, policy?.errorFields?.changeOwner, policyID]);
+
+    const confirm = useCallback(() => {
+        if (error === CONST.POLICY.OWNERSHIP_ERRORS.HAS_FAILED_SETTLEMENTS || error === CONST.POLICY.OWNERSHIP_ERRORS.FAILED_TO_CLEAR_BALANCE) {
+            // cannot transfer ownership if there are failed settlements, or we cannot clear the balance
+            Navigation.navigate(ROUTES.WORKSPACE_MEMBER_DETAILS.getRoute(policyID, accountID));
+        }
+
+        const ownershipChecks = WorkspaceSettingsUtils.getOwnershipChecks(error);
+
+        PolicyActions.requestWorkspaceOwnerChange(policyID, ownershipChecks);
+        Navigation.dismissModal();
```

Comment:
I don't think we should dismiss this modal. We should keep it visible and show a loading indicator while the request is pending